src/compiler/binder.ts

```diff
@@ -3631,9 +3631,13 @@ namespace ts {
                 break;
 
             case SyntaxKind.NumericLiteral:
-                if ((<NumericLiteral>node).numericLiteralFlags & TokenFlags.BinaryOrOctalSpecifier) {
+                const flags = (<NumericLiteral>node).numericLiteralFlags;
+                if (flags & TokenFlags.BinaryOrOctalSpecifier) {
                     transformFlags |= TransformFlags.AssertES2015;
                 }
+                if (flags & TokenFlags.BigInt) {

```

Comment:
I feel like instead of putting this in `NumericLiteral`, you should have a new `BigIntLiteral` type; not sure if others on the team agree with me here.

Reply:
It was easier to parse them both as `SyntaxKind.NumericLiteral` expressions since many of the same flags still apply (namely `HexSpecifier`, `BinarySpecifier`, `OctalSpecifier`, `ContainsSeparator`). I split them once they were converted into `LiteralType`s.

Reply:
I think i agree with @DanielRosenwasser .. Though the flags are same, you could create the SyntaxKind.BigIntLiteral making it easier to ensure they arent treated like numbers unknowingly. (note the type of number and bigint is different and hence explicit token kind is justified)